haystack/document_stores/elasticsearch/base.py

```diff
@@ -206,6 +291,24 @@ def _create_document_index(self, index_name: str, headers: Optional[Dict[str, st
             if not self._index_exists(index_name, headers=headers):
                 raise e
 
+    def _create_embedding_field_mapping(self):
+        mapping = {"type": "dense_vector", "dims": self.embedding_dim}
+        if self.index_type == "exact":
+            return mapping
+        mapping["index"] = True
+        mapping["similarity"] = self._get_similarity_string()
+        return mapping
```

Comment:
Let's add separate methods for `es8.ElasticsearchDocumentStore` and `es7.ElasticsearchDocumentStore`.